pkg/cmd/release/create/http.go

```diff
@@ -174,6 +178,26 @@ func createRelease(httpClient *http.Client, repo ghrepo.Interface, params map[st
 	}
 	defer resp.Body.Close()
 
+	// This code checks if we received a 404 while attempting to create a release without
+	// the workflow scope, and if so, returns an error message that explains a possible
+	// solution to the user.
+	//
+	// If the same file (with both the same path and contents) exists
+	// on another branch in the repo, releases with workflow file changes can be
+	// created without the workflow scope. Otherwise, the workflow scope is
+	// required to create the release.
+	//
+	// https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes
+	if resp.StatusCode == http.StatusNotFound && tokenMissingWorkflowScope(resp) {
+		normalizedHostname := ghauth.NormalizeHostname(resp.Request.URL.Hostname())
+		errMissingRequiredWorkflowScope := errors.New(heredoc.Docf(`
+				HTTP 404: Failed to create release, "workflow" scope may be required
+				To request it, run gh auth refresh -h %[1]s -s workflow
+			`, normalizedHostname))
+
+		return nil, errMissingRequiredWorkflowScope
+	}
```

Comment:
This technically works though I don't know how much this should align to other places we  have to do similar errors:

https://github.com/cli/cli/blob/1ec23f698f50f709a54099aba633d0031d21d6b4/pkg/cmd/gpg-key/add/add.go#L87-L91

https://github.com/cli/cli/blob/1ec23f698f50f709a54099aba633d0031d21d6b4/pkg/cmd/gpg-key/list/list.go#L60-L65

https://github.com/cli/cli/blob/1ec23f698f50f709a54099aba633d0031d21d6b4/pkg/cmd/project/shared/queries/queries.go#L1461-L1466

https://github.com/cli/cli/blob/1ec23f698f50f709a54099aba633d0031d21d6b4/pkg/cmd/auth/status/status.go#L47-L54

To summarize:

1. Should we print the error here and return a silent error?
2. Should we emphasize the command to run?
3. Should we avoid the `HTTP 404` in the error message?

🤔

Reply:
> Should we print the error here and return a silent error?

What is the benefit for returning this silent error?

> Should we emphasize the command to run?

Sure, I like that idea.

> Should we avoid the HTTP 404 in the error message?

Yes, I can remove it.

Reply:
> > Should we print the error here and return a silent error?
> 
> What is the benefit for returning this silent error?

Of course, the benefit is that we can print the error right here instead of passing an `error` which is eventually printed 🤪 

> > Should we emphasize the command to run?
> 
> Sure, I like that idea.
> > Should we avoid the HTTP 404 in the error message?
> 
> Yes, I can remove it.

Done 👍 

----

I made these changes and a refactor in deb34d6456d14efe73ccd61fcb88e063da5f4794

Also, thank you for the examples @andyfeller ✨ - they were very helpful.